### ISSUE TITLE
Nextflow Kuberun doesn't see public github repository

### ISSUE THREAD

**Issue by kulagins** [nextflow-io/nextflow#1848]
### Expected behavior and actual behavior

I try doing `nextflow kuberun nextflow-io/rnatoy -v task-pv-claim:/mnt/data`, where nextflow-io/rnatoy is a sample nextflow project in github and `task-pv-claim` is the name of the persistent volume claim I created.

Expected: the project should be found and should run.
Actual: The github repository is not being found.

But there is a repository exactly at this address. 
### Steps to reproduce the problem
* Have a minikube kubernetes config with 1 Node
* Install nextflow by doing `curl -s https://get.nextflow.io | bash` and moving the nextflow file to a  `$PATH`-location.
* Create a Persistent Volume and Persistent Volume Claim as described [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-persistent-volume-storage/)
* (optional) Run rnatoy without kuberun, as described [here](https://github.com/nextflow-io/rnatoy)
* Launch the workflow with kuernetes with `nextflow kuberun nextflow-io/rnatoy -v task-pv-claim:/mnt/data` , where `task-pv-claim` is the name of the persistent volume claim that you created.

### Program output 

```Pod started: small-hopper
N E X T F L O W  ~  version 20.10.0
Pulling nextflow-io/rnatoy ...
WARN: Cannot read project manifest -- Cause: api.github.com
Cannot find `nextflow-io/rnatoy` -- Make sure exists a GitHub repository at this address `https://github.com/nextflow-io/rnatoy`
``` 
Same output when I insert the complete path to the repository.
### Environment 

* Nextflow version: 20.10.0.5430
* Java version: 11.0.9.1
* Operating system: Ubuntu 20.04
* Bash version: Version 5.0.17(1)-release (x86_64-pc-linux-gnu)

### Additional context

Thank you for any input and have a greate New Year!

**Comment by kulagins** [comment]
Some additional findings.
A helloworld application does not work either: same output as described when I do  `nextflow kuberun nextflow-io/hello -v task-pv-claim:/mnt/data`
However, when I run it with docker, it works:
```
$ nextflow run nextflow-io/rnatoy -with-docker
N E X T F L O W  ~  version 20.10.0
Launching `nextflow-io/rnatoy` [pedantic_williams] - revision: 5ec6d657cc [master]
R N A T O Y   P I P E L I N E    
=============================
genome: /home/kulaginasv/.nextflow/assets/nextflow-io/rnatoy/data/ggal/ggal_1_48850000_49020000.Ggal71.500bpflank.fa
annot : /home/kulaginasv/.nextflow/assets/nextflow-io/rnatoy/data/ggal/ggal_1_48850000_49020000.bed.gff
reads : /home/kulaginasv/.nextflow/assets/nextflow-io/rnatoy/data/ggal/*_{1,2}.fq
outdir: results
executor >  local (5)
[3d/c2a77b] process > buildIndex (ggal_1_48850000_49020000.Ggal71.500bpflank) [100%] 1 of 1 ✔
[2e/87653a] process > mapping (ggal_liver)                                    [100%] 2 of 2 ✔
[c3/463853] process > makeTranscript (ggal_gut)                               [100%] 2 of 2 ✔
Done!
```
The image also seems to have been downloaded already:
```
nextflow pull nextflow-io/rnatoy
Checking nextflow-io/rnatoy ...
Already-up-to-date - revision: 5ec6d657cc [master]
```

**Comment by pditommaso** [comment]
I suspect it may be related to some incompatibility with your shared storage. You may want to check if deploying the execution via https://tower.nf solve or help to troubleshoot the problem.

**Comment by kulagins** [comment]
Thank you, Paolo! I tried the tower.nf, but I really need to run this on my local machine. I am attempting to write some custom kubernetes components and need to test them this way.
So I fixed this issue with my minikube setup by doing the following:
* Check if you have more than one mode `kubectl get nodes`
* If you do, delete it with `minikube node delete %NODENAME%`. It is likely to be something like `minikube-04` 
*  Restart the minikube (not sure if required, but I did that)
* Recreating the persistent volume and persistent volume claim is not necessary
* Give your default serivce account privileges to start pods:
 ``` 
kubectl create rolebinding default-edit --clusterrole=edit --serviceaccount=default:default
kubectl create rolebinding default-view --clusterrole=view --serviceaccount=default:default
 ```
And it worked for me.
